### ISSUE TITLE
feature request? GROUP BY (or similar)

### ISSUE THREAD

**Issue by duralog** [arangodb/arangodb#765]
I understand that there is the `COLLECT` command, however I would like to be able to do this:

```
FOR data IN posts
  GROUP BY data.uid [DEPTH [offset, limit]]
  SORT bp.date DESC
```

this would normally, without the `DEPTH` option, return an array of the latest post each user has made. however, with the `DEPTH` option, it would make arrays for each of the unique data.uid, and put DEPTH number of elements inside of the array

```
FOR data IN posts
  GROUP BY data.uid DEPTH 5
  SORT bp.date DESC
```

...

```
[
  [
    {"_id": "posts/1234", "uid": 11, "text": "yooo", "date": 1392557991934},
    {"_id": "posts/1235", "uid": 11, "text": "hey there", "date": 1392557990934}
  ],
  [
    {"_id": "posts/1236", "uid": 13, "text": "word up", "date": 1392557991934},
    {"_id": "posts/1237", "uid": 13, "text": "jaja", "date": 1392557990934},
    {"_id": "posts/1337", "uid": 13, "text": "last bit of spam", "date": 1392557970934}
  ]
]
```

I'm just iterating over the list of uids and running a bunch of queries then putting them into an array right now (which is fine for testing). I just found out about user functions too - but haven't tried yet, so maybe this could be implemented that way. however, I probably will be doing this operation often and would want it optimized by indices. I also I found out I could do this is making a batch query and iterating over the uid but I'm using Foxx and that's the same as just doing a bunch of queries -- so then, I thought I could do it with subqueries, but I haven't figured out how to do yet.

P.S. also, in foxx, it'd be nice to have some sort of async way to run a bunch of queries in parallel as well, but that's for another time
